### DISCUSSION TITLE
Sorting by stars would be helpful

### DISCUSSION THREAD

**Discussion by ConfusedVorlon** [SwiftPackageIndex/SwiftPackageIndex-Server#1803] [Ideas]
It would be nice to sort by stars (as opposed to filtering)

I just did a search for a package to handle the keychain, and number of stars looks like a really strong signal for a quick answer on which one I should use.

**Comment by daveverwer** [comment]
I agree, and I think some sort options would be good for those "package list" views (not only the search results but packages from an author and various other lists we have).

Thanks for the suggestion, Rob!

**Reply by Sherlouk** [reply]
Do you remember the reason why we stopped sorting my score (which, among other things, is a factor of stars)?

**Reply by finestructure** [reply]
I don’t think we did stop. At least I don’t remember removing it!

**Reply by Sherlouk** [reply]
ahh, just double checked, I think I was remembering author pages which are sorted differently to main search results.

**Reply by finestructure** [reply]
Yes, I believe those are alphabetical.

**Comment by ConfusedVorlon** [comment]
Throwing in a random thought here - I think it is good to sort by recency **by default**. 

That gives newer packages a chance to at least be seen.

**Comment by Wallby** [comment]
Is this feature planned? Not having a way to sort search results is very inconvenient. There might be a bunch of smaller packages by independent authors, but for many projects, stable more well-known packages are what I'd be looking for.

**Reply by daveverwer** [reply]
It would still be useful, of course, but it’s not currently planned for implementation. However, you can work around for your use case here with filters. For example, searching for `keychain stars:>50` will only show packages with more than 50 stars, which should surface the more established packages you’re looking for while keeping results ranked by relevance.

You can also combine filters, so `keychain stars:>50 last_activity:>=2025-01-01` would narrow it further to actively maintained packages.

**Reply by Wallby** [reply]
Well problem is that sometimes high rated package are ones advertised well, not per se built well. So starts don't work so well in my experience. Actual times downloaded works well, but it also differs per use case. What might be a lot of downloads for a small library is not for a bigger one.
